### PR TITLE
Flatten /err

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

Continuing the flatten by flattening the `stan/math/{prim,rev,fwd,mix}/*/err` folder to `stan/math/{prim,rev,fwd,mix}/err`. Only 2 files (check_ordered and check_finite) needed actual merging. The rest of the changes are just fixes of includes, removing the /scal/,/mat/ or /arr/ parts.

## Tests

/

## Side Effects

/

## Checklist

- [x] Math issue #937 

- [x] Copyright holder: Rok Češnovar

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
This also merges in changes from #1545 that touch the /err folder. So this will be ready for review once that is merged to develop.

**Comment by rok-cesnovar** [comment]
This PR is running on Jenkins with the branch of the Stan PR https://github.com/stan-dev/stan/pull/2872
These 2 PRs need to be merged together to keep develop stable.

File: stan/math/prim/err/check_pos_semidefinite.hpp [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-#include <stan/math/prim/mat/err/check_symmetric.hpp>
+#include <stan/math/prim/err/check_symmetric.hpp>
 #include <stan/math/prim/mat/fun/Eigen.hpp>
 #include <stan/math/prim/mat/fun/value_of_rec.hpp>
-#include <stan/math/prim/scal/err/throw_domain_error.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
-#include <stan/math/prim/scal/err/check_positive.hpp>
+#include <stan/math/prim/err/throw_domain_error.hpp>
+#include <stan/math/prim/err/check_not_nan.hpp>
+#include <stan/math/prim/err/check_positive.hpp>
```
Comment (mcol) [inline stan/math/prim/err/check_pos_semidefinite.hpp]: Mind reordering?

File: stan/math/prim/err/check_ordered.hpp [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+#define STAN_MATH_PRIM_ERR_CHECK_ORDERED_HPP
 
 #include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err/throw_domain_error.hpp>
 #include <stan/math/prim/mat/fun/Eigen.hpp>
-#include <stan/math/prim/scal/err/throw_domain_error.hpp>
-#include <sstream>
 #include <vector>
+#include <sstream>
 #include <string>
```
Comment (mcol) [inline stan/math/prim/err/check_ordered.hpp]: Ordering.

File: stan/math/prim/err/is_corr_matrix.hpp [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 #include <stan/math/prim/mat/fun/Eigen.hpp>
-#include <stan/math/prim/mat/err/is_pos_definite.hpp>
-#include <stan/math/prim/mat/err/is_symmetric.hpp>
-#include <stan/math/prim/scal/err/is_positive.hpp>
-#include <stan/math/prim/scal/err/is_size_match.hpp>
+#include <stan/math/prim/err/is_pos_definite.hpp>
+#include <stan/math/prim/err/is_symmetric.hpp>
+#include <stan/math/prim/err/is_positive.hpp>
+#include <stan/math/prim/err/is_size_match.hpp>
 #include <cmath>
```
Comment (mcol) [inline stan/math/prim/err/is_corr_matrix.hpp]: Ordering.

File: stan/math/prim/err/check_3F2_converges.hpp [hunk 4]
```diff
@@ -2,7 +2,7 @@
 #define STAN_MATH_PRIM_SCAL_ERR_CHECK_3F2_CONVERGES_HPP
```
[the remark this answers is not in the document]
Comment (rok-cesnovar) [reply stan/math/prim/err/check_3F2_converges.hpp]: Yeah I somehow missed fixing the PRIM_SCAL ones. Thanks for catching that!

**Pull request merged**